applications/minotari_app_utilities/src/identity_management.rs

```diff
@@ -121,11 +136,37 @@ pub fn setup_node_identity<P: AsRef<Path>>(
 ///
 /// ## Returns
 /// Result containing a NodeIdentity on success, string indicates the reason on failure
-fn load_node_identity<P: AsRef<Path>>(path: P) -> Result<NodeIdentity, IdentityError> {
+fn load_node_identity<P: AsRef<Path>>(path: P, transport_type: TransportType) -> Result<NodeIdentity, IdentityError> {
     check_identity_file(&path)?;
 
     let id_str = fs::read_to_string(path.as_ref())?;
     let id = json5::from_str::<NodeIdentity>(&id_str)?;
+
+    let id = if transport_type == TransportType::Tcp {
+        let current_addresses = id.public_addresses();
+        debug!(
+            target: LOG_TARGET,
+            "Filtering addresses for TCP transport. Current addresses: {:?}",
+            current_addresses
+        );
+        // For TCP transport remove all onion addresses
+        let filtered_addresses: Vec<Multiaddr> = current_addresses
+            .into_iter()
+            .filter(|addr| !addr.iter().any(|p| matches!(p, Protocol::Onion3(_))))
+            .collect();
+        debug!(
+            target: LOG_TARGET,
+            "After filtering for TCP transport, {} addresses remain: {:?}",
+            filtered_addresses.len(),
+            filtered_addresses
+        );
+        id.set_public_addresses(filtered_addresses);
+        save_as_json(&path, &id)?;
+        id
+    } else {
+        id
+    };
```

Comment:
Let's assume we start up tcp from a previous mixed tcp/tor type id fle, then we change the type to tcp and save the file, then we start again as tor, then the id file does not contain tor info anymore? How will the tor inof be refreshed?